dfget/core/core.go

```diff
@@ -209,6 +200,40 @@ func downloadFile(cfg *config.Config, supernodeAPI api.SupernodeAPI,
 	return err
 }
 
+func doDownload(cfg *config.Config, supernodeAPI api.SupernodeAPI,
+	register regist.SupernodeRegister, result *regist.RegisterResult, timeout time.Duration) error {
+	var getter downloader.Downloader
+	isBackDownload := false
+	if cfg.BackSourceReason > 0 {
+		getter = backDown.NewBackDownloader(cfg, result)
+		isBackDownload = true
+	} else {
+		printer.Printf("start download by dragonfly...")
+		getter = p2pDown.NewP2PDownloader(cfg, supernodeAPI, register, result)
+	}
+
+	err := downloader.DoDownloadTimeout(getter, timeout)
+	if err == nil {
```

Comment:
if err := ...; err == nil {}

Reply:
I don't think so. And the err will be used later.